src/pudl/transform/eia860.py

```diff
@@ -46,29 +51,38 @@ def ownership(eia860_dfs, eia860_transformed_dfs):
     # The fix we're making here is only known to be valid for 2011 -- if we
     # get older data... then we need to to revisit the cleaning function and
     # make sure it also applies to those earlier years.
-    if (min(o_df.report_date.dt.year)
+    if (min(own_df.report_date.dt.year)
             < min(pc.working_partitions['eia860']['years'])):
         raise ValueError(
             f"EIA 860 transform step is only known to work for "
             f"year {min(pc.working_partitions['eia860']['years'])} and later, "
-            f"but found data from year {min(o_df.report_date.dt.year)}."
+            f"but found data from year {min(own_df.report_date.dt.year)}."
         )
 
     # Prior to 2012, ownership was reported as a percentage, rather than
     # as a proportion, so we need to divide those values by 100.
-    o_df.loc[o_df.report_date.dt.year < 2012, 'fraction_owned'] = \
-        o_df.loc[o_df.report_date.dt.year < 2012, 'fraction_owned'] / 100
-
-    o_df = (
-        o_df.astype({
-            "owner_utility_id_eia": pd.Int64Dtype(),
-            "utility_id_eia": pd.Int64Dtype(),
-            "plant_id_eia": pd.Int64Dtype(),
-            "owner_state": pd.StringDtype()
-        })
-    )
+    own_df.loc[own_df.report_date.dt.year < 2012, 'fraction_owned'] = \
+        own_df.loc[own_df.report_date.dt.year < 2012, 'fraction_owned'] / 100
+
+    own_pk = [
+        'report_date',
+        'plant_id_eia',
+        'generator_id',
+        'owner_utility_id_eia'
+    ]
+    dupes_to_drop = own_df[own_df.duplicated(subset=own_pk)]
+    dupes_to_drop = dupes_to_drop[
+        dupes_to_drop.plant_id_eia.isin(OWNERSHIP_DUPLICATE_PLANT_GEN_IDS)]
+    own_df = own_df.drop(dupes_to_drop.index)
+
+    own_df = own_df.astype({
+        "owner_utility_id_eia": pd.Int64Dtype(),
+        "utility_id_eia": pd.Int64Dtype(),
+        "plant_id_eia": pd.Int64Dtype(),
+        "owner_state": pd.StringDtype()
+    })
```

Comment:
I know you didn't add these but I think we can delete them because the types will be applied uniformly in the post transform step

Reply:
I don't love them either, but I'd like to not touch them as part of this PR, just in case they do mater -- and instead integrate this kind of type casting cleanup into the integration of the new metadata / harvesting system.

Reply:
Uuuugh, okay actually (per our chat) I need to drop all but one copy of the duplicates, not drop one of the duplicates. And add an assertion here that there are no more duplicates thereafter. And also add some inline comments explaining wtf is happening here.